README.md

```diff
@@ -10,8 +10,90 @@ The following resources are available:
 - dnsmadeeasy_record
 - dnsmadeeasy_record_A
 
-> NOTE: this cookbook is work in progress
+## Usage
 
+### `dnsmadeeasy_credentials_file`
 
+You can create a credentials file on a remote server, if you need to perform DNS operations outside of Chef.
+
+Here is how:
+
+```ruby
+# We need to set our API key/secret somewhere.
+# Future versions will support encrypted data bag.
+
+@api_key    = '134324....239f9120f'
+@api_secret = 'afa998....1ffg99999'
+@encr_key   = ENV['ENCRYPTION_KEY']
+
+dnsmadeeasy_credentials_file 'my-corp' do
+  api_key @api_key
+  api_secret @api_secret
+  encryption_key @encr_key
+  action :create
+end
+```
+
+### `dnsmadeeasy_record_A`
+
+Use this to create, update, or delete an A record.
+
+```ruby
+# Let's assign this server hostname 'app001.awesome-corp.com':
+
+dnsmadeeasy_record_A 'app001' do
+  domain 'awesome-corp.com'
+  value node['ipaddress']
+  api_key @api_key
+  api_secret @api_secret
+  encryption_key @encr_key
+  ttl 90
+  action :create
+end
+```
+
+### `dnsmadeeasy_record_CNAME`
+
+This resource creates/updates/deletes CNAMEs to other hosts:
+
+```ruby
+# Let's add this server's name to the CNAME www.
+# This would only work, if server's name has been added as a valid A record 
+# also.
+dnsmadeeasy_record_CNAME www do
+  domain 'awesomne-corp.com'
+  value node['name']
```

Comment:
Is this right? Should this be `node['ipaddress]'`? If not, how is associating the node's IP address with the `www` CNAME?

Reply:
So CNAME maps an existing hostname to an IP address. Since this is a README example, I am just throwing it there expecting it to exist. This would fail if the hostname does not exist.

What this can be used for, is — say we build 6 load balancers. Each will have a unique hostname, such as `lb001 — lb006`, but they will all also have a CNAME associated with each lb: `www -> [ lb001, lb002., .. ]`